### PR TITLE
fix: fix sidebar scrolling and adaptation for mobile

### PR DESCRIPTION
**Pull request by ihor-romaniuk**
**TL;DR -**

This pull request contains minor fixes related to the responsiveness of the blocks for mobile phones. All the before and after screenshots are below.

**Related Pull Requests**
PR to the open-release/palm.master branch: https://github.com/openedx/frontend-app-gradebook/pull/362
PR to the open-release/quince.master branch: https://github.com/openedx/frontend-app-gradebook/pull/370

**What changed?**
- fixed text overload in the title when the course ID is too long

|  Before |  After |
|---|---|
|  image |  image |

- fixed appearance and scrolling of the sidebar

|  Before |  After |
|---|---|
|  image |  image |
|   image  |   image |

- Fixed adaptation of the search and filter buttons on mobile phones.

|  Before |  After |
|---|---|
|  image |  image |

- removed excess empty space under the main table of contents

|  Before |  After |
|---|---|
|  image |  image |

- fix adaptation modal content to mobile view

|  Before |  After |
|---|---|
|  image  |  image |

**Developer Checklist**
- [ ] Test suites passing
- [ ] Documentation and test plan updated, if applicable
- [ ] Received code-owner approving review
- [ ] Bumped version number [package.json](../package.json)

**Reviewer Checklist**

Collectively, these should be completed by reviewers of this PR:

- [ ] I've done a visual code review
- [ ] I've tested the new functionality

FYI: @openedx/content-aurora

### CODE REVIEW

**Comment by openedx-webhooks** [comment]
Thanks for the pull request, @ihor-romaniuk!  Please note that it may take us up to several weeks or months to complete a review and merge your PR.

Feel free to add as much of the following information to the ticket as you can:
- supporting documentation
- [Open edX discussion forum threads](https://discuss.openedx.org/)
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).


Please let us know once your PR is ready for our review and all tests are green.

**Comment by mphilbrick211** [comment]
Hi @arbrandes! Could you take a look at this and the 2 backports? PR to the open-release/palm.master branch: https://github.com/openedx/frontend-app-gradebook/pull/362
PR to the open-release/quince.master branch: https://github.com/openedx/frontend-app-gradebook/pull/370

**Review by arbrandes — APPROVED** [review]
Looks great, thank you!

**Comment by openedx-webhooks** [comment]
@ihor-romaniuk 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/frontend-app-gradebook&entry.867055334=https://github.com/openedx/frontend-app-gradebook/pull/363&entry.1484655318=https://github.com/ihor-romaniuk&entry.752974735=2023-10-23+13:14&entry.1917517419=2024-04-03+18:27&entry.2133058324=Yes) so we can improve your experience in the future.
